pkg/alertmanager/alertmanager.go

```diff
@@ -319,12 +325,21 @@ func (am *Alertmanager) Stop() {
 		am.dispatcher.Stop()
 	}
 
+	if service, ok := am.state.(*state); ok {
```

Comment:
This seemed like the least-worst option for the time being but open to suggestions, alternatives being:
- Add Stop() method to State interface - Means we would need to a Stop method to Peer in Prometheus
- Storing the service separately as `stateService` or something, again.